src/v/storage/offset_to_filepos.cc

```diff
@@ -226,6 +250,11 @@ ss::future<result<offset_to_file_pos_result>> convert_end_offset_to_file_pos(
 
                     return batch_consumer::consume_result::accept_batch;
                 }
+
```

Comment:
When I was reading this code yesterday, my thought was that maybe an entire batch was skipped. Is that possible, and if so, could we get here?

Reply:
Looking at the compaction reducer, it does seem possible that batches are skipped completely https://github.com/redpanda-data/redpanda/blob/dev/src/v/storage/compaction_reducers.cc#L213

It seems like batches with 0 records can be dropped, this seems to match @VladLazar 's proposal that the batch containing the offset `5913` might have been dropped.

If the batch is missing, then this callback won't be called, but some other batch afterwards in the segment will still have been processed, as we did not get the `Segment does not contain searched for offset` error.